semtypes/src/main/java/io/ballerina/types/BddPath.java

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.types;
+
+import io.ballerina.types.subtypedata.BddAllOrNothing;
+import io.ballerina.types.subtypedata.BddNode;
+import io.ballerina.types.typeops.BddCommonOps;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *  Represents path from root to leaf (ending with true).
+ *  bdd gets the Bdd for this path
+ *
+ * @since 3.0.0
+ */
+public class BddPath {
+    Bdd bdd;
+    List<Atom> pos;
+    List<Atom> neg;
+
+    private BddPath(BddPath bddPath) {
+        this.bdd = bddPath.bdd;
+        this.pos = bddPath.pos;
```

Comment:
```suggestion
        this.pos = new ArrayList<>(bddPath.pos); // pos: path.pos.clone()
```
same for neg
We need to create a new list here so that when we add to this pos, it doesn't get added to the original list